TypeCasting.md

```diff
@@ -0,0 +1,42 @@
+# Type Casting
+
+In general, follow the [official guidelines](https://docs.swift.org/swift-book/LanguageGuide/TypeCasting.html).
+
+## Convention - Static casting
+Use keyword `as` for static casting. Only use static casting when automatic type casting is not available or when the compiler's inferred type does not match the intended type.
+
+## Rationale
+Compiler can infer most types correctly so static casting is not necessary in most cases.
+
+## Examples 
+
+``` swift
+// Bad: unnecessary type casting
+let padding = 42 as CGFloat
```

Comment:
doesn't this need to be as! to compile?

Reply:
Nope, it's a constant so the type can be interpreted by the compiler. No force casting is needed.